voluncheer/opportunityboard/models.py

```diff
@@ -114,3 +117,47 @@ def duration(self):
         duration = end_datetime - start_datetime
         duration_seconds = dt.timedelta(seconds=duration.seconds)  # remove date
         return duration_seconds
+
+
+class ArchivedOpportunity(models.Model):
```

Comment:
Duplicating all of these fields would not be my choice for strategy. You could create a class that inherits from Opportunity above or, better yet, add the necessary fields to support archiving to the Opportunity class above.

Reply:
Right. I totally agree with Joe on this.

Reply:
I've tried to inherit the opportunity model, however, when creating an Archive object it will also create an Opportunity(like Volunteer to User). Hence it will create duplicate object to opportunityboard and profile. I think that will cause more issues.

Reply:
OK, using a set of fields to identify archived opportunities would have been my preference anyway.